src/main/java/net/dv8tion/jda/api/interactions/components/LayoutComponent.java

```diff
@@ -45,6 +45,28 @@
     @Nonnull
     List<ItemComponent> getComponents();
 
+    /**
+     * Whether this LayoutComponent is compatible with {@link net.dv8tion.jda.api.entities.Message Messages}.
+     * <br>This will return false if this LayoutComponent contains components that are incompatible with Messages.
+     *
+     * @return True, if all components of this LayoutComponent are compatible with messages.
+     */
+    default boolean isMessageCompatible()
```

Comment:
These should also check if the layout type itself is compatible, there might be new layouts in the future which are only compatible with messages or modals. This can be added to the type enum as well.